src/coreclr/src/jit/lclvars.cpp

```diff
@@ -1983,16 +2001,20 @@ bool Compiler::StructPromotionHelper::ShouldPromoteStructVar(unsigned lclNum)
 #if FEATURE_MULTIREG_STRUCT_PROMOTE
         // Is this a variable holding a value with exactly two fields passed in
         // multiple registers?
-        if ((structPromotionInfo.fieldCnt != 2) && compiler->lvaIsMultiregStruct(varDsc, compiler->info.compIsVarArgs))
+        if (compiler->lvaIsMultiregStruct(varDsc, compiler->info.compIsVarArgs))
         {
-            JITDUMP("Not promoting multireg struct local V%02u, because lvIsParam is true and #fields != 2\n", lclNum);
-            shouldPromote = false;
+            if ((structPromotionInfo.fieldCnt != 2) &&
+                !((structPromotionInfo.fieldCnt == 1) && varTypeIsSIMD(structPromotionInfo.fields[0].fldType)))
+            {
+                JITDUMP("Not promoting multireg struct local V%02u, because lvIsParam is true and #fields != 2\n",
```

Comment:
```suggestion
                JITDUMP("Not promoting multireg struct local V%02u, because lvIsParam is true and #fields != 2 and not a single SIMD?\n",
```

Reply:
Done - with a minor rewording